pkg/datasource.go

```diff
@@ -376,9 +546,9 @@ func (o *OCIDatasource) getCompartments(ctx context.Context, region string, root
 
 	var page *string
 	reg := common.StringToRegion(region)
```

Comment:
Based on my testing this logic to set the region needs to be before the `GetTenancy()` call above. Otherwise in the following scenario the template variables won't be settable by the user:

1. User imports an existing dashboard with all of the metrics plugin template variables defined.
2. When the dashboard is loaded, the region template variable will default to the first entry in the list which is `af-johannesburg-1`
3. Unless the configured tenancy is setup to actually use OCI services in the `af-johannesburg-1` region (which won't be the case for most users), the region for the identityClient variable will be set to `af-johannesburg-1`.
4. But if the tenancy doesn't have that region enabled the call to get compartments will fail.
5. Even if the user changes the region selection on the dashboard to be an appropriate region where the tenancy has resources, none of the template variable selections will update since the `GetTenancy()` call will continue to use the Johannesburg tenancy where the user does not have access.

Reply:
Good. I modified accordingly and it seems to work.